tests/python/unittest/test_operator.py

```diff
@@ -3496,6 +3496,10 @@ def test_special_functions_using_scipy():
     mathematical_core("gammaln", lambda x: mx.sym.gammaln(x), lambda x: scipy_special.gammaln(x),
                      lambda x: scipy_special.psi(x), 0.5, 0.5)
 
+    # erf
+    mathematical_core("erf", lambda x: mx.sym.erf(x), lambda x: scipy_special.erf(x),
+                     lambda x: 2.0 / math.sqrt(math.pi) * math.exp(-(x ** 2)), 0.5, 0.5)
```

Comment:
also in this method in line 3488 when importing scipy fails, we are skipping these tests. Shouldn't we be failing them by raising an error/exception?

Reply:
scipy is optional dep and CI is set up to test both with and without scipy.

Reply:
the same applies to sparse support

Reply:
ok